spec/services/catalog_search_service_spec.rb

```diff
@@ -12,5 +12,40 @@
   end
 
   it { expect(service).to be_an AbstractSearchService }
-  it { expect(service.search(query)).to be_an CatalogSearchService::Response }
+
+  describe 'Response' do
+    it { expect(service.search(query)).to be_an CatalogSearchService::Response }
+
+    describe '#additional_facet_details' do
+      before do
+        expect_any_instance_of(CatalogSearchService::Response).to receive(:facets).at_least(:once).and_return(facet_data)
+      end
+
+      let(:addl_facet_details) { service.search(query).additional_facet_details('blah') }
+
+      context 'when the facet data is present' do
+        let(:facet_data) { [{ 'name' => 'format_main_ssim', 'items' => [{ 'label' => 'Database', 'hits' => 123 }]}] }
+
+        it 'has hits returned from the facet' do
+          expect(addl_facet_details.hits).to eq 123
+        end
+
+        it 'has a href that links to the full query' do
+          expect(addl_facet_details.href).to include('/?q=blah&f[format_main_ssim][]=Database')
+        end
+      end
+
+      context 'when the facet does not have any hits' do
+        let(:facet_data) { [{ 'name' => 'format_main_ssim', 'items' => [{ 'label' => 'Database', 'hits' => '0' }]}] }
```

Comment:
should this be `'hits' => 0` to match the Solr response?

Reply:
I guess this was kind of a hidden test for string numbers are cast to integers.  Happy to change this or make that test more explicit.

Reply:
ok that's fine